docs/proposals/27-falco-extension.md

```diff
@@ -0,0 +1,352 @@
+to the `extensions` section of their shoot manifest:
+
+```
+  - type: shoot-falco-service
+    providerConfig:
+      apiVersion: falco.extensions.gardener.cloud/v1alpha1
+      kind: FalcoServiceConfig
+      # optional, will use the latest version tagged as "supported"
+      falcoVersion: 0.37.0
+      # optional, will always default to true
+      autoUpdate: true|false
+      # optional, "gardener" or "falcoctl", will default to "gardener"
+      resources: gardener
+      gardener:
+        # optional, defaults to true
+        useFalcoRules: true
+        # optional, defaults to false
+        useFalcoIncubatingRules: false
+        # optional, defaults to false
+        useFalcoSandboxRules: false
+        # references configmaps in project namespace containing custom rules
+        ruleRefs:
+        - ref: rules1
+      falcoctl:
+        # Falcoctl configuration as defined in 
+        # https://github.com/falcosecurity/charts/blob/master/charts/falco/values.yaml#L366
+      # publish Falco events to other event store (optional)
+      customWebhook:
+        enabled: true|false
+        address: <http...>
+        customHeaders: ...
+        # check if ssl certificate of the output is valid
+        checkCerts: ...
+```
+
+In a simple setup the following default configuration is sufficient to
+deploy Falco:
+
+```
+  - providerConfig:
+      apiVersion: falco.extensions.gardener.cloud/v1alpha1
+      kind: FalcoServiceConfig
+    type: shoot-falco-service
+```
+
+In this case the extension will deploy the current Falco version and will 
+take care of automated updates. The ruleset will be managed by Gardener and 
+will only provide the standard rules which are designed to offer basic 
+detection capabilities but to avoid too many false positives. Visualization 
+of events will be provided in the Gardener dashboard.
+
+This will correspond to the following configuration which will be reflected
+in the shoot manifest:
+
+```
+  - providerConfig:
+      apiVersion: falco.extensions.gardener.cloud/v1alpha1
+      kind: FalcoServiceConfig
+      falcoVersion: <latest Falco version marked as "supported">
+      falcoAutoUpdate: true
+      resources: gardener
+      gardener:
+        useFalcoRules: true
+        useFalcoIncubatingRules: false
+        useFalcoSandboxRules: false
+    type: shoot-falco-service
+```
+
+Falco versions are maintained by the extension in the same way operating
+system-, and Kubernetes versions are maintained by Gardener. If no 
+version is specified the Falco extension will select the currently supported
+version.
+
+There are two methods for managing rules and plugins: either use setup 
+provided by Gardener by setting `resources` to `gardener` or use 
+`falcoctl` by setting `resources` to `falcoctl`.
+
+Custom rules in the Gardener setup can be stored in a ConfigMap and applied 
+to the same project as the cluster. The ConfigMap must be configured as a 
+resource in the cluster as described in [6]:
```

Comment:
It is very nice to have the rules defined in a separate resource instead of in-lining them in the shoot spec, this way they can be easily re-used by multiple shoots. 

What I am missing is how the resource is referred from the extension spec, I do not see the `customRules` resource to be referred there.